Automattic/wp-calypso
# Editor Stats - track core global styles updates & saves

#### Changes proposed in this Pull Request

Adds tracking for global styles update and save events (`wpcom_block_editor_global_styles_update` and `wpcom_block_editor_global_styles_save`).  Unfortunately, core Global Styles seems to update through `useEntityProp` (which in turn uses `updateEntityRecord`), which requires passing the entire content object of Global Styles and not only the items that have changed.  This means if we want to track individual changes we need to compare the updated content object to the original content object.

Here we attempt to dissect the changes made to `wp_global_styles` and map them into events with the following properties:
* `block_type` - the block name if the change is to an individual block style or setting.
* `section` - the type of style being updated (color, typography, etc.)
* `field` - the property being changed (text, background, fontSize, fontFamily, etc.)
* `field_value` - the value of the property being changed.  This is currently populated as `reset` if the field is removed. (is this better than just not having a field value in this case?)
* `element_type` - The element type the setting corresponds to.  Currently I only see this as `link`, but given the structure of the content object I think its important we leave this property as case other items are added in this way (span, etc?).
* `palette_slug` - Slug of the color palette value that was updated.

Some examples of how these event properties populate in GS update events (note any global properties such as blog_id, editor_type, etc. are also included):

* Changing text color: `{ section: "color", field: "text", field_value: "var:preset|color|green" }`
* Changing background color: `{ section: "color", field: "background", field_value: "var:preset|color|green" }`
* Changing a typography setting: `{ section: "typography", field: "fontSize", field_value: "20px" }`
* Returning to 'default' on a setting: `{ section: "typography", field: "fontSize", field_value: "reset" }`
* Changing link text color: `{ element_type: "link", section: "color", field: "text",  field_value: "var:preset|color|green" }`
* Changing a color palette setting: `{ section: "color", field: "palette", field_value: "#3ec71a", palette_slug: "green"}`

Performing any of these actions at the individual block level will also add the `block_type` property (ex `block_type: "core/buttons"`)

#### Testing instructions



* Follow steps to enable tracking debugging at PCYsg-nrf-p2
* Open the site editor, change various global styles options and verify the `wpcom_block_editor_global_styles_update` tracks events follow the format noted above.
* Save the global styles changes and verify the `wpcom_block_editor_global_styles_save` tracks events follow the format above.



Related to https://github.com/Automattic/wp-calypso/issues/53410
